src/App.tsx

```diff
@@ -1,26 +1,171 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable */
+import React, { useEffect, useState, useMemo, useRef } from 'react';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Todo } from './types/Todo';
+import {getTodos, addTodo, USER_ID, deleteTodo, updateTodo,} from './api/todos';
+import { ErrorTypes } from './types/ErrorTypes';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes>(ErrorTypes.Empty);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(FilterStatus.All,);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const inputAddRef = useRef<HTMLInputElement>(null);
+
+  const filteredTodos = useMemo(
+    () =>
+      todos.filter(todo => {
+        if (filterStatus === FilterStatus.All) {
+          return true;
+        }
+
+        return filterStatus === FilterStatus.Completed
+          ? todo.completed
+          : !todo.completed;
+      }),
+    [todos, filterStatus],
+  );
+
+  const todosLeftNum = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+  const todosCompletedNum = useMemo(
+    () => todos.filter(todo => todo.completed).length,
+    [todos],
+  );
+  const areAllTodosCompleted = useMemo(
+    () => todos.length === todosCompletedNum,
+    [todos],
+  );
+  const handleAddTodo = async (todoTitle: string) => {
+    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
```

Comment:
don't forget about indents
```suggestion
  const todosLeftNum = useMemo(
    () => todos.filter(todo => !todo.completed).length,
    [todos],
  );
  
  const todosCompletedNum = useMemo(
    () => todos.filter(todo => todo.completed).length,
    [todos],
  );
  
  const areAllTodosCompleted = useMemo(
    () => todos.length === todosCompletedNum,
    [todos],
  );
  
  const handleAddTodo = async (todoTitle: string) => {
    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
    
    try {
      const newTodo = await addTodo({ title: todoTitle, completed: false });
```